modin/core/execution/ray/common/engine_wrapper.py

```diff
@@ -108,6 +111,14 @@ def put(cls, data, **kwargs):
         ray.ObjectID
             Ray object identifier to get the value by.
         """
+        if isinstance(data, FunctionType):
+            qname = data.__qualname__
+            if "<locals>" not in qname and "<lambda>" not in qname:
+                ref = cls._func_cache.get(data, None)
```

Comment:
Could there be a key collision?

Reply:
Do you mean a hash table collision? It's possible, like for any other keys, but unlikely. All the functions have a different hash code:
```python
>>> hash(pandas.DataFrame.add)
8744465041078
>>> hash(pandas.DataFrame.mul)
8744465041118
```
Even, in case of collisions, I believe it's not critical, because the hash table is quite small.